torch/utils/data/datapipes/utils/common.py

```diff
@@ -16,6 +18,34 @@
 ]
 
 
+def validate_input_col(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
+
+    Args:
+        fn: The function to check.
+        input_col: The input column to check.
+    Returns:
+        None.
+    Raises:
+        TypeError: If the function is not compatible with the input column.
+    """
+    sig = inspect.signature(fn)
+    if isinstance(input_col, (list, tuple)):
+        sz = len(input_col)
+    else:
+        sz = 1
+
+    if len(sig.parameters) > sz:
```

Comment:
Thanks to @NivekT's review, it triggers me thinking we might need to raise Exception when `len(sig.parameters) < sz` as well.
And, could you please add the corresponding tests?

Reply:
Resolved in https://github.com/pytorch/pytorch/pull/79344/commits/97bbd73dc3c9c415c1f305efd6450e6889d07a3b